pkg/resource/deploy/builtins.go

```diff
@@ -98,7 +98,7 @@ func (p *builtinProvider) Diff(urn resource.URN, id resource.ID, state, inputs r
 	if !inputs["name"].DeepEquals(state["name"]) {
 		return plugin.DiffResult{
 			Changes:     plugin.DiffSome,
-			ReplaceKeys: []resource.PropertyKey{"name"},
+			ChangedKeys: []resource.PropertyKey{"name"},
```

Comment:
This seems long overdue, changing a stack reference's source is more akin to changing config than a delete-and-replace. I'm surprised we haven't seen more issues with folks surprised by cascading replaces from stack ref changes.

Reply:
Curious - the customer issue had me thinking this was a stack ref change that caused a provider replacement, causing resource replacements.

Reply:
It shouldn't, our replacement logic doesn't work like that, I'd err towards Pat's comments leave this as is and investigate the customer issue some more

Reply:
The customer was able to re-create their stack after the fact.  I suspect the actual issue was not the stackref itself changing, but the underlying environment replacing providers, and making changes in more than one cluster.  I would _guess_ they created a new namespace in a second cluster, and were unable to delete the existing namespace since it existed in a different cluster.  I could be wrong, but the customer was able to find a workaround here.  I'm happy to close this PR, though, if this is not our preferred behavior here.